ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java

```diff
@@ -158,7 +173,11 @@ public Response streamQuery(
     CommandStoreUtil.httpWaitForCommandSequenceNumber(
         commandQueue, request, commandQueueCatchupTimeout);
 
-    return handleStatement(securityContext, request, statement);
+    return handleStatement(securityContext, request, statement, startTime);
+  }
+
+  public void closeMetrics() {
+    pullQueryMetrics.ifPresent(PullQueryExecutorMetrics::close);
```

Comment:
There needs to be a null check here as pulQueryMetrics can be null and this will cause a NPE on shutdown if it wasn't configured. (I have seen this NPE in real life)